src/qt/listctrl.cpp

```diff
@@ -417,17 +442,37 @@ class wxQtListModel : public QAbstractTableModel
         const int row = static_cast<int>(info.GetId());
         const int col = info.m_col;
 
-        wxCHECK_MSG( static_cast<size_t>(row) < m_rows.size(),
-            false, "Invalid row");
-        wxCHECK_MSG( static_cast<size_t>(col) < m_headers.size(),
-            false, "Invalid col");
+        wxCHECK_MSG( row < rowCount(), false, "Invalid row");
+        wxCHECK_MSG( col < columnCount(), false, "Invalid col");
 
         const QModelIndex modelIndex = index(row, col);
+        QVector<int> roles;
+
+        if ( info.m_mask & wxLIST_MASK_STATE )
+        {
+            if ( (info.m_stateMask & wxLIST_STATE_FOCUSED) &&
+                (info.m_state & wxLIST_STATE_FOCUSED) )
+                m_view->setCurrentIndex(modelIndex);
+            if ( info.m_stateMask & wxLIST_STATE_SELECTED )
+            {
+                QItemSelectionModel *selection = m_view->selectionModel();
+                const QItemSelectionModel::SelectionFlag flag =
+                    info.m_state & wxLIST_STATE_SELECTED
+                        ? QItemSelectionModel::Select
+                        : QItemSelectionModel::Deselect;
+                selection->select(modelIndex,  flag|QItemSelectionModel::Rows);
+            }
+        }
+
+        if ( IsVirtual() )
+        {
+            // wxLIST_MASK_STATE is the only mask supported by virtual wxListCtrl
+            return true;
+        }
+
         RowItem &rowItem = m_rows[row];
         ColumnItem &columnItem = rowItem[col];
 
-        QVector<int> roles;
-
```

Comment:
Not sure why was this moved above, there doesn't seem to be any reason for doing this?